tmt/steps/report/reportportal.py

```diff
@@ -39,112 +36,169 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
     """
 
     _data_class = ReportReportPortalData
 
+    def handle_response(self, response: requests.Response) -> None:
+        if not response.ok:
+            raise tmt.utils.ReportError(f"\n\
+Received non-ok status code from ReportPortal. Response text:\n{response.text}")
+        else:
+            self.debug(f"Response code from the server: {response.status_code}")
+            self.debug(f"Message from the server: {response.text}")
+
     def go(self) -> None:
         """
-        Read executed tests, prepare junit, compress it to a zip zile and
-        send it to the ReportPortal instance.
+        Create a ReportPortal launch and its test items,
+        fill it with all parts needed and report the logs.
         """
 
         super().go()
 
-        # Check the data, show interesting info to the user
-        # Required fields are: url, token and project
         server = self.get("url")
+        server = server.rstrip("/")
         if not server:
             raise tmt.utils.ReportError("No ReportPortal server url provided.")
-        server = server.rstrip("/")
+
+        project = self.get("project")
+        if not project:
+            raise tmt.utils.ReportError("No ReportPortal project provided.")
 
         token = self.get("token")
         if not token:
             raise tmt.utils.ReportError("No ReportPortal token provided.")
 
-        project = self.get("project")
-        if not project:
-            raise tmt.utils.ReportError("No ReportPortal project provided.")
-        self.info("project", project, color="green")
-
-        # Use provided launch name, default to run workdir name
-        assert self.step.plan.my_run is not None
-        assert self.step.plan.my_run.workdir is not None
-        launch_name = self.get("launch-name") or self.step.plan.my_run.workdir.name
-        self.info("launch", launch_name, color="green")
-
-        # Generate a xUnit report
-        import_junit_xml()
-        assert junit_xml is not None
-        suite = junit.make_junit_xml(self)
-        data = junit_xml.TestSuite.to_xml_string([suite])
-
-        # Zip the report
-        bytestream = io.BytesIO()
-        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,
-                             compresslevel=1) as zipstream:
-            # XML file names are irrelevant to ReportPortal
-            with zipstream.open("tests.xml", "w") as entry:
-                entry.write(data.encode("utf-8"))
-        bytestream.seek(0)
-
-        # Send the report to the ReportPortal instance
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
+        url = f"{server}/api/{version}/{project}"
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-vars") or []
```

Comment:
@4N0body5 
Can you give an example for fmf? 
env-vars multi usage and their format?  

report:
    how: reportportal
    url: https://reportportal-rhivos.apps.ocp-c1.prod.psi.redhat.com
    token: TOKEN
    project: TEST_PROJECT
    launch-name: hirte-tests
    env-vars: key: distro value: fedora
    env-vars:  arch: aarch63

Reply:
Setup:
Token and url can be exported in environment variables .
Project name is a mandatory argument.
Launch name is optional (for identifying all launches=plans with given name), otherwise each launch is named after plan by default (launch per plan).
Argument env-vars is only an optional argument for list of variables that happen to be an exception obeying the rule that does not allow to show "TMT_*" variables.

Example for setup:
```
export TMT_REPORT_REPORTPORTAL_URL=https://reportportal-rhivos.apps.ocp-c1.prod.psi.redhat.com/
export TMT_REPORT_REPORTPORTAL_TOKEN=<your-token-from-RP-user-profile>
```

Example for arguments:
```
report:
    how: reportportal
    project: <project-name>
    env-vars: [TMT_TEST_DATA,TMT_PLAN_DATA]
```

Example for plan.fmf:
```
summary: Basic test plan for report to RP
description: Testing plan data for reportportal test

discover:
    how: fmf
execute:
    how: tmt

context:
    component: tmt
    distro: rhel-8
    arch: x86_64
    compose: RHEL-8.9.0-20230323.20
    subsystem: xx
    purpose: upgrade
    milestone: rc

environment:
    RELEASE: rhel8
    example_env_var: example_val
```

Reply:
Ack

Reply:
You're right, more specifically here: https://tmt.readthedocs.io/en/latest/spec/context.html#spec-context
It shall be listed in plan fmf data as I show in an example above and it does allow to add any other metadata with same syntax.

Reply:
Yes, checked it, you are correct, 
Context could be passed as attributes,

Reply:
Related to environment vars all uploaded to tests properties
I do not under stand the use case for 
env-var under the report, I do not see it uploaded anywhere